Lib/unittest/test/testmock/testhelpers.py

```diff
@@ -871,6 +872,19 @@ def test_autospec_on_bound_builtin_function(self):
         mocked.assert_called_once_with(4, 5, 6)
 
 
+    def test_autospec_getattr_partial_function(self):
+        # bpo-32153 : getattr returning partial functions without
+        # __name__ should not create AttributeError in create_autospec
+        class Foo(object):
+
+            def __getattr__(self, name):
+                return partial(lambda name: name, name)
```

Comment:
What is `name` here? The parameter of `__getattr__` or the parameter of the lambda? It is better to use different names for avoiding confusion.

Unconditional return in `__getattr__()` can cause issues, because internal code can get some special instance attributes and be confused by getting a lambda. Return only for specific name, and raise an AttributeError in other cases.

```python
if name == '...':
    return ...
raise AttributeError
```

Reply:
> What is name here? The parameter of __getattr__ or the parameter of the lambda? It is better to use different names for avoiding confusion.

I will change them. Thanks.

> Unconditional return in __getattr__() can cause issues, because internal code can get some special instance attributes and be confused by getting a lambda. Return only for specific name, and raise an AttributeError in other cases.

From the [original report](https://bugs.python.org/issue32153#msg307115) this is about `__getattr__` returning a partial function unconditionally for **all** attributes and partial functions have no `__name__` attribute during copy which occurs at https://github.com/python/cpython/blob/master/Lib/unittest/mock.py#L108. I am slightly confused over why we should have an if clause for this and raise `AttributeError` that might change the original report that needs to be tested.